src/assign.c

```diff
@@ -428,6 +428,10 @@ SEXP assign(SEXP dt, SEXP rows, SEXP cols, SEXP newcolnames, SEXP values)
   }
   if (any_duplicated(cols,FALSE)) error(_("Can't assign to the same column twice in the same query (duplicates detected)."));
   if (!isNull(newcolnames) && !isString(newcolnames)) error(_("newcolnames is supplied but isn't a character vector"));
+  if (Rf_inherits(values, "POSIXlt")) {
+    warning(_("Values of type POSIXlt detected and converted to POSIXct. We do not recommend the use of POSIXlt at all because it uses 40+ bytes to store one date. Use as.POSIXct() to avoid this warning."));
```

Comment:
compare with warning from data.table()

POSIXlt column type detected and converted to POSIXct. We do not recommend use of POSIXlt at all because it uses 40 bytes to store one date.

Can the two warnings be made exactly the same? (easier for translators probably)

Reply:
> (easier for translators probably)

Possibly, but this is thrown from C while the other is thrown from R, so currently they live in different domains (`R-data.table.pot` vs `data.table.pot`), so a given translator might not even know about the two different messages. I think there are two ways to align them:

 1. Make the R message thrown from a C wrapper like `posixlt_warning = function() .Call(Cposixlt_warning)`
 2. Call `gettext(., domain="data.table")` on the _exact_ string used in C.

The former adds a little clutter to our .so, while the latter is a bit fragile by depending on the exact wording. I'm not sure either is better, or if either is worth investing in in the first place.

Reply:
Not sure the effort to align them is worth, if it is not ensured that both are translated by the same translator (or by the same translation string).